core/os/file/path.go

```diff
@@ -88,6 +89,16 @@ func (p Path) IsDir() bool {
 	return info != nil && info.IsDir()
 }
 
+// IsExecutable returns true if the file at p.value exists, is executable
+// and is not a directory.
+func (p Path) IsExecutable() bool {
+	if runtime.GOOS == "windows" {
+		return true
+	}
+	info := p.Info()
+	return info != nil && !info.IsDir() && (info.Mode().Perm()&0111) != 0
```

Comment:
I was going to ask you to update the doc comment to say that on Windows it always returns true, but then realized we should be a bit better and do this:

```suggestion
	info := p.Info()
	if info == nil || info.IsDir() {
		return false
	}
	return runtime.GOOS == "windows" || (info.Mode().Perm()&0111) != 0
```

Sorry for the back-and-forth. Please do update the comment to indicate that on Windows all files are considered executable.